### DISCUSSION TITLE
How to work on customer to create their own Products, Order

### DISCUSSION THREAD

**Discussion by bbtronald** [vendurehq/vendure#1187] [General]
Do i really need to create a plugin for this? to let the customers create their own shop and at the same time create the channel for multi tenancy. But how can i create a List of Products, is there a way to copy the code from the admin-api and put it in shop-api. So that Customers can manage their own Products, Orders, Products Categories. 

Give me an idea gus how to do it.

**Comment by michaelbromley** [comment]
It depends on your exact requirements but generally yes, you'll probably want to encapsulate this extended functionality in a plugin.

If you want to duplicate some of the Admin API functionality in the Shop API, I suggest you basically copy-paste those resolvers into your own plugin code. Resolvers are usually very "thin" - most of the code is contained in the service layer, which you can re-use.

**Reply by bbtronald** [reply]
Thank you for the Fast Reply appreciate it.

@michaelbromley Our Requirements is multi tenant where the owner of the store can manage their own Products, Catalog, Orders, Sales, Their own Marketplace. where they can atleast post Product Sales, then Customers can add to cart from different store, can update he's own cart. Can checkout. Basically like AliExpress.


`If you want to duplicate some of the Admin API functionality in the Shop API, I suggest you basically copy-paste those resolvers into your own plugin code. Resolvers are usually very "thin" - most of the code is contained in the service layer, which you can re-use.`

Is there a guide to do this, Can you point me out where it is the documentation? what im seeing is the typescript-api services
https://www.vendure.io/docs/typescript-api/services/product-service/
In which im still looking for how to implement this into the plugin.

**Reply by michaelbromley** [reply]
Docs on creating your own resolvers in plugins: https://www.vendure.io/docs/plugins/plugin-examples/extending-graphql-api/ - but also read the whole plugin section to get more context.
